confdgnmi/testtool/gNMI_Interface/SubscribeLibrary.py

```diff
@@ -1,18 +1,95 @@
 from __future__ import annotations
 
-from confd_gnmi_common import make_gnmi_path
+import typing as t
+
+from confd_gnmi_common import make_gnmi_path, encoding_str_to_int
 from confd_gnmi_client import ConfDgNMIClient
-from gNMIRobotLibrary import gNMIRobotLibrary
+from CapabilitiesLibrary import CapabilitiesLibrary
+
+import gnmi_pb2 as gnmi
+
+import threading
+import queue
+
+SlistType = t.Optional[t.Union[gnmi.Poll, gnmi.SubscriptionList]]
+
+
+class Requester(threading.Thread):
+    def __init__(self, client: ConfDgNMIClient) -> None:
+        self.client: ConfDgNMIClient = client
+        self.slist_queue: queue.Queue[SlistType] = queue.Queue()
+        self.response_queue: queue.Queue[gnmi.SubscribeResponse] = queue.Queue()
+        super().__init__()
+
+    def run(self) -> None:
+        # TODO: this dumps the MultiThreaded thing exception for XR
+        for response in self.client.stub.Subscribe(self.requests(),
+                                                   metadata=self.client.metadata):
+            self.response_queue.put(response)
+
+    def requests(self) -> t.Iterator[gnmi.SubscribeRequest]:
+        while (slitem := self.slist_queue.get()) is not None:
+            if isinstance(slitem, gnmi.SubscriptionList):
+                yield gnmi.SubscribeRequest(subscribe=slitem)
+            elif isinstance(slitem, gnmi.Poll):
+                yield gnmi.SubscribeRequest(poll=slitem)
 
+    def enqueue(self, item: SlistType) -> None:
+        self.slist_queue.put(item)
 
-class SubscribeLibrary(gNMIRobotLibrary):
+    def responses(self, wait: bool = True) -> t.Iterator[gnmi.SubscribeResponse]:
+        try:
+            while (response := self.response_queue.get(block=wait)) is not None:
+                yield response
+        except queue.Empty:
+            pass
```

Comment:
Is there equivalent for  `responses.cancel()` found in `read_subscribe_responses` in the client?